crates/matrix-sdk/src/deduplicating_handler.rs

```diff
@@ -24,7 +24,20 @@ use tokio::sync::Mutex;
 
 use crate::{Error, Result};
 
-type DeduplicatedRequestMap<Key> = Mutex<BTreeMap<Key, Arc<Mutex<Result<(), ()>>>>>;
+/// State machine for the state of a query deduplicated by the
+/// [`DeduplicatingHandler`].
+enum QueryState {
+    /// The query hasn't completed yet. This doesn't mean it hasn't *started*
+    /// yet, but rather that it couldn't get to completion: some
+    /// intermediate steps might have run.
+    NotFinishedYet,
```

Comment:
`NotFinishedYet` sounds like we should wait and see if it'll get finished next time, if we're sure that this only happens only if we cancel the future, perhaps `Cancelled` would be a better name.